hivemq-extension-sdk/src/main/java/com/hivemq/extension/sdk/api/packets/puback/ModifiablePubackPacket.java

```diff
@@ -0,0 +1,49 @@
+package com.hivemq.extension.sdk.api.packets.puback;
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extension.sdk.api.annotations.Nullable;
+import com.hivemq.extension.sdk.api.packets.general.ModifiableUserProperties;
+import com.hivemq.extension.sdk.api.packets.general.UserProperties;
+import com.hivemq.extension.sdk.api.packets.publish.AckReasonCode;
+
+/**
+ * A {@link PubackPacket} that can be modified before it is sent to the client.
+ *
+ * @author Yannick Weber
+ */
+public interface ModifiablePubackPacket extends PubackPacket {
+
+    /**
+     * Set a {@link AckReasonCode} to the PUBACK packet.
+     * <p>
+     * Switching from successful to unsuccessful and vice versa is not supported.
```

Comment:
What happens if i still do it?

Reply:
@throws IllegalStateException If switching from successful reason code to unsuccessful reason code or vice versa.

I should add this to the documentation text.